openHAB 5.2 release blog post (#564)
openhab/openhab-website: blog/2026-07-05-openhab-5-2-release.md

---
layout: BlogPost
category: blog
title: openHAB 5.2 Release
author: Florian Hotze and others
date: "2026-07-05T20:00:00+01:00"
previewimage: /uploads/2026-07-05-openhab-5-2-release/header.webp
tags:
  - news
excerpt: >-
  Can you believe it's already been a year since the major release of openHAB 5.0? We are thrilled to introduce openHAB 5.2, packed with exciting new features, plenty of bug fixes and amazing new AI features!
---

Can you believe it's already been a year since the major release of openHAB 5.0?
We are thrilled to introduce openHAB 5.2, packed with exciting new features, plenty of bug fixes, and amazing new AI features!

But that's not all!
Following the release of [openHAB 5.1](/blog/2025-12-22-openhab-5-1-release.html) last winter with many "under the hood" changes, this release focuses on more visual changes.

> This release includes only a small number of breaking changes, so upgrading from openHAB 5.2 should be really smooth.

Reviewer: either "to 5.2" or "from 5.1".

But that's not all!
Following the release of [openHAB 5.1](/blog/2025-12-22-openhab-5-1-release.html) last winter with many "under the hood" changes, this release focuses on more user-facing changes.

This release includes only a small number of breaking changes, so upgrading from openHAB 5.1 should be really smooth.
Please review our [**official release notes**](https://github.com/openhab/openhab-distro/releases/tag/5.2.0) to learn about all breaking changes, new add-ons, enhancements, and fixes.

With that being said, we again want to share some statistics that show the progress in numbers and highlight some of the new features that you should not miss.

# Activity

<!-- https://github.com/florian-h05/github-stats?tab=readme-ov-file#contribution_stats -->
<!-- count changed lines: https://github.com/openhab/openhab-core/compare/5.1.0...5.2.0.RC2 -->
<!-- count total lines with cloc: https://github.com/AlDanial/cloc -->

Every new feature in openHAB 5.2 is a testament to the dedication of our vibrant, global community.
Activity across our GitHub repositories has been non-stop over these past few months.

> Altogether, 141 contributors poured their efforts into a whopping 3.082 commits across our [GitHub repositories](https://github.com/openhab).

Reviewer 2: ```suggestion
Altogether, 141 contributors poured their efforts into a whopping 3,082 commits across our [GitHub repositories](https://github.com/openhab).
```

This big release includes impactful work across every part of the ecosystem:

- **[Core](https://github.com/openhab/openhab-core)**: 323 pull requests, including 110 enhancements and 58 bug fixes — adding 44.791 lines of new code.

> - **[Add-ons](https://github.com/openhab/openhab-addons)**: 798 pull requests, including 223 enhancements and 233 bug fixes, resulting in 554.128 lines of new code.

Reviewer 2: ```suggestion
- **[Add-ons](https://github.com/openhab/openhab-addons)**: 798 pull requests, including 223 enhancements and 233 bug fixes, resulting in 554,128 lines of new code.
```

> - **[UIs](https://github.com/openhab/openhab-webui)**: 467 pull requests, including 108 enhancements and 175 bug fixes, totaling in 37.915 lines of new code.

Reviewer 2: ```suggestion
- **[UIs](https://github.com/openhab/openhab-webui)**: 467 pull requests, including 108 enhancements and 175 bug fixes, totaling in 37,915 lines of new code.
```

Countless [other repositories](https://github.com/openhab) — from documentation and automation helper libraries to our mobile apps — also received a steady stream of contributions too extensive to list individually.

A massive thank you to everyone driving openHAB forward!

We also want to take the moment to shine a spotlight on our top code contributors for this release,
recognized for landing the most commits in their respective repositories:

<!-- https://github.com/florian-h05/github-stats?tab=readme-ov-file#contribution_stats -->

- Holger Friedrich ([@holgerfriedrich](https://github.com/holgerfriedrich)) for openHAB Core & Add-ons
- Laurent Garnier ([@lolodomo](https://github.com/lolodomo)) for openHAB Core
- Leo Siepel ([@lsiepel](https://github.com/lsiepel)) for openHAB Add-ons
- Jacob Laursen ([@jlaur](https://github.com/jlaur)) for openHAB Add-ons

> - Florian Hotze ([@florian-h05](https://github.com/florian-h05)) for the openHAB Main UI & openHAB Core

Reviewer 3: You are considering the number of PRs here ?
I did not have in mind that I contributed so many PRs in core repo (47).
If number of lines was considered, I guess that others are bigger contributors (@mherwege or @Nadahar for examples I have in mind - as I reviewed some of their contributions).

Author: I am counting commits using a custom script to load that data from the GitHub API:

```
      "top_contributors_by_commits": {
        "holgerfriedrich": 57,
        "lolodomo": 46,
        "florian-h05": 40,
        "dilyanpalauzov": 36,
        "dependabot[bot]": 32
      },
```

Author: The GitHub UI gets worse and worse for the insights IMO, for openHAB 5.1 it was possible to customize the queried timerange by modifying the HTTP query params, now that doesn't work anymore ...

But wrt to additions, the ranking between 3 Jan and 28 Jun is:

1. Nadahar (16075)
2. florian-h05 (12716)
3. mherwege (11345)
4. lolodomo (10987)
5. kaikreuzer (6781)

So let's add Ravi Nadahar and Mark Herwege to the ranking and remove the reference to most commits ...

You can read more about our new MCP binding, including tool use, client setup, and example use cases at the [MCP integration](/addons/integrations/mcp/) add-on page.

## Core Runtime Enhancements

### YAML Configuration Files for the UI

_Jimmy Tanagra ([@jimtng](https://github.com/jimtng)), openHAB Maintainer_

In this release, file-based YAML support has been introduced for UI pages and custom widgets.

Previously, customized Main UI components were strictly created and managed via Main UI.
With this update, you can now define, organize, and back up your pages and widgets directly inside your `$OPENHAB_CONF` directory using simple, clean YAML files.

Key details of this new feature include:

- **Marketplace Integration:**<br/>
  Custom widgets can be easily adapted from the openHAB Marketplace for your configuration files.

>   You simply need to use the UID as the map key and nest the marketplace configuration underneath it.

Reviewer 2: ```suggestion
- **Marketplace Integration:**<br/>
  Custom widgets can be easily adapted from the openHAB Marketplace for your configuration files. 
  With the updated marketplace support, widgets using the new YAML format (which includes top-level `version:` and `widgets:` sections) can be copied directly without any changes.
  For older marketplace widgets (identified by a top-level `uid:` key), you will need to add the widget's UID as a map key under your file's top-level `widgets:` section and nest the marketplace configuration underneath it.
```

- **Broad Page Support:**<br/>
  The YAML parser natively supports standard page layouts (Layout, Tabs, Map, Floor Plan, and Chart).
   Additionally, support has been extended to seamlessly handle file-based home page configurations and the Overview page layout.

This change completes an essential missing link for power users who want to back up, version-control (via Git), and deploy their entire openHAB user interface side-by-side with their items, things, and rules.

#### Syntax

The precise syntax rules, structure, and configuration keys are documented in the [Pages YAML Documentation](/docs/configuration/yaml/pages.html) and [Widgets YAML Documentation](/docs/configuration/yaml/widgets.html).

If you have existing UI-managed components that you wish to mirror or move into text files, you can easily view their structure directly within the UI's **Code** tab.
To streamline this process, a **Copy** button has been added to the page and widget lists.
By selecting one or more items, you can instantly copy the correct file-ready YAML syntax directly to your clipboard.

### YAML File Format for Rules & Rule Templates

> TODO @Nadahar

Reviewer: ```suggestion
openHAB 5.2 supports [rules](/docs/configuration/yaml/rules.html) and [rule templates](/docs/configuration/yaml/ruletemplates.html) in YAML files.

In Main UI the **Code** tab for the rules page has been changed to work similarly to those for Things and Items.
The Code tab now contains **file‑compatible syntax** for both YAML and DSL formats.
Not all rules can be expressed in all formats.
This is evaluated on a per-rule basis, and only the applicable formats are shown. If a rule can't be expressed in neither YAML nor DSL, the Code tab will be hidden.

Rules can also be copied to the clipboard in either YAML or DSL format both from the rules page itself, or from the rules overview page. By using the rules overview page, it's possible to copy the definition for many rules at once, but only the rules that can be expressed in the target format.

Rule templates don't have Main UI editing support, so there's no visible change for them, but they can be exported to YAML by using the `/file-format/ruletemplates` endpoint in the REST API, for example using the API Explorer.

Rule templates, widgets and block libraries have also gained support for the new YAML format on the community marketplace.
```

Reviewer: I don't know if you want to split this into "core" and "Main UI" sections, feel free to cut and mix as appropriate.

Reviewer: Also, I just hate hype and bragging, so I have problems using the "marketing tone" used elsewhere. Feel free to "spice it up", it's fine by me as long as I don't have to write it.

Author: The motivation behin the blog post is to highlight what's new - not marketing.

Reviewer: Sure, I'm just saying that I find it "difficult" to express myself in the same way as a lot of what's here, so if you think it's "too boring", feel free to alter the text.

- **More Sophisticated Tooltips:**<br/>
  Chart tooltips have been improved to display the unit of measurement or mapped Item states if defined through state descriptions.
- **Extended UI Support for Configuration:**<br/>
  It has always been possible to customize charts by providing ECharts options through the code tab.
  openHAB 5.2 finally adds UI support for configuring the most common series options, including:
  - color
  - series label position (for line/bar charts)
  - border radius (for bar charts)
  - x-axis style (label only, label & line, label & line & axis ticks)
  - y-axis style (axis line, minor lines, areas can be individually enabled or disabled)
- **Initial Period for Fixed Period Charts:**<br/>
  The initial period for fixed period charts can now be set.
- **Fixed Chart Periods over 1 Year:**<br/>
  Fixed chart periods 2 years, 3 years, and five years have been added, extended the chart range.
- **New Yearly Aggregation:**<br/>
  A new yearly aggregation has been added, allowing to compare several years in fixed period charts.

### Redesigned Persistence Configuration

> TODO @mherwerge/@jsjames

Reviewer 4: Configuring persistence through the UI has been streamlined, making it easier to understand and configure.
The main entry point is now the configuration for a specific persistence service. From that configuration you can easily configure the items, strategies and filters. You can pick existing strategies and filters or create new ones.
The cron strategy editor now fully supports updating an existing cron expression.
Related to this enhancement, we also enhanced the health checks on the persistence configurations, warning you for more inconsistencies or conflicts.

### Persistence Status on the Item page

_Mark Herwege ([@mherwege](https://github.com/mherwege)), openHAB Maintainer_

> TODO @mherwege

Reviewer 4: Did you ever wonder if a specific item is being persisted? Do you want to know what service is in play? What the persistence rules are?
Until now, you had to build a mental model looking at your persistence configuration and interpret all rules to find out. It was not always easy to understand what happens.
This enhancement does all of that for you and makes it visible on the Item page in the UI.

### Setup Wizard

_Mark Herwege ([@mherwege](https://github.com/mherwege)), openHAB Maintainer_

> TODO @mherwege

Reviewer 4: The setup wizard is the first thing a user sees when setting up a new system. This release improves that experience in a number of ways:
- added new configuration steps, improving the converage of the initial wizard configuration.
- added steps to briefly explain openHAB concepts in the context of an initial configuration (can be disabled).
- the wizard keeps track of what steps have already been done.
- the ability to go back and forward.
- breadcrumb display with ability to jump to a specific step.
- link from the UI to open the wizard again, so you can come back to it at any time.
The wizard will always keep an already existing configuration. Changes are only adding to that.

Reviewer 3: > link from the UI to open the wizard again, so you can come back to it at any time.

Please mention where as it is well hidden.

Reviewer 4: It is in the same page as where you select the theme for the UI I believe. I can´t check, writing this on my phone without access to OH.

Reviewer 3: Yes at the bottom of the About page

```yaml
# Generated output: $OPENHAB_CONF/yaml/composed/main.yaml
version: 1

things:
  mqtt:topic:livingroom-lamp:
    bridge: mqtt:broker:main
    channels:
      power:
        type: switch
        config:
          stateTopic: livingroom-lamp/state
          commandTopic: livingroom-lamp/set/state

  mqtt:topic:kitchen-coffee:
    bridge: mqtt:broker:main
    channels:
      power:
        type: switch
        config:
          stateTopic: kitchen-coffee/state
          commandTopic: kitchen-coffee/set/state

  mqtt:topic:bedroom-heater:
    bridge: mqtt:broker:main
    channels:
      power:
        type: switch
        config:
          stateTopic: bedroom-heater/state
          commandTopic: bedroom-heater/set/state

items:
  Livingroom_Lamp_Power:
    type: Switch
    label: Livingroom Lamp Power
    channel: mqtt:topic:livingroom-lamp:power

  Kitchen_Coffee_Power:
    type: Switch
    label: Kitchen Coffee Power
    channel: mqtt:topic:kitchen-coffee:power

  Bedroom_Heater_Power:
    type: Switch
    label: Bedroom Heated Blanket Power
    channel: mqtt:topic:bedroom-heater:power
```

</details>

### JavaScript & Python Debugger Support

> ### Home Connect Direct Binding

Reviewer 5: Suggested content for the Home Connect Direct Binding section:

```suggestion
### Home Connect Direct Binding

_Jonas Brüstel ([@bruestel](https://github.com/bruestel))_

In this release, the new Home Connect Direct Binding integrates [Home Connect](https://www.home-connect.com/) enabled appliances into openHAB over your local network.
It works with the many brands built on the Home Connect platform, including Bosch, Siemens, Gaggenau, NEFF, and more.
Unlike the existing Home Connect binding, this implementation talks directly to your appliances on the LAN.
There is no cloud account and no internet connection involved during operation, which results in reliable, low latency control and keeps working even when the manufacturer's cloud is unavailable.

A broad range of appliance types is supported, including dishwashers, washers, dryers, washer/dryer combinations, ovens, warming drawers, coffee makers, cook processors (Cookit), hoods, cooktops, and fridge/freezers.
Appliances are found automatically on your network via mDNS and show up in the Inbox, so adding them is as simple as accepting the discovery result.

#### Rich channel support

Each appliance type comes with a set of preconfigured channels for common functions such as power state, door contact, operation state, remaining program time, program progress, and program control.
Many channels are added dynamically based on the exact features your appliance model reports, for example the different oven cavities, i-Dos dosing on washers, or the individual cooling compartments of a fridge/freezer.
Beyond that, custom channels let you expose any internal value or capability from the appliance profile, so even functions that are not covered by the standard channels remain accessible.

#### A console for setup and diagnostics

The binding ships with its own built in web console.
It lists all your appliances with their connection status and hardware details, and provides a real time monitoring view of every message exchanged between openHAB and an appliance.
A handy openHAB configuration generator produces ready to use YAML or DSL snippets for custom channels directly from the values you see in the live message stream.
```

Author: Many thanks! I've added your content to the blog post.

Rules DSL has been a core part of openHAB since its initial release many years ago.
In openHAB 5.2, this text-based rule engine receives a major update designed to improve flexibility, modularity, and feature parity with standard rules.

Whether you write rules manually in text files or manage them via the Main UI, this release removes several historical limitations and introduces new ways to structure your Rules DSL-based automation:

- **Native Conditions:**<br/>
  openHAB 5.2 adds support for native condition blocks as known from Main UI.
  You can now cleanly separate conditions from actions, removing the need to clutter the action with if conditions and guard clauses:
- **Triggerless Rules:**<br/>
  You can now define a DSL rule without triggers, which is perfect for creating reusable scripts meant to be executed manually or called exclusively by other rules.
- **Convenience Functionality:**<br/>
  The standard library has been extended with a number of new functions, allowing for more concise and readable code.
  Refer to [Rules DSL: Convenience Functionality](/docs/configuration/rules-dsl.html#convenience-functionality) for more details.
- **Inter-Rule Execution:**<br/>

>   Rules DSL now officially supports invoking and executing another rule from within a rule.

Reviewer 6: I would skip the word `officially` - without it the meaning is the same

Here a new bullet could be added, possibly with a different wording, possibly extending the next example:

* Global variables, defined before the first rule definition, can use previously defined global variables.

Examples are available at https://github.com/openhab/openhab-core/pull/5086 .

Reviewer 6: Of course, the above is only valid for Rules from files,  not for rules from Main UI.  This should be spelled explicitly, as to alter for this item the caption of the list: “Whether you write rules manually in text files or manage them via the Main UI, this release removes several historical limitations…”

### YAML Configuration Files for the UI

_Jimmy Tanagra ([@jimtng](https://github.com/jimtng)), openHAB Maintainer_

In this release, file-based YAML support has been introduced for UI pages and custom widgets.

Previously, customized Main UI components were strictly created and managed via Main UI.

> With this update, you can now define, organize, and back up your pages and widgets directly inside your `$OPENHAB_CONF` directory using simple, clean YAML files.

Reviewer: Block libraries support has also been added, even if it was done by me instead of @jimtng. I don't see any point in making a separate section for that, so it should be mentioned here.

That said, the documentation for using them is lacking, so I'm not sure how many will figure it out from the documentation.  But, the code from the "Code tab" can be pasted directly into YAML files, which should make it easy to "export" existing UI widgets at east.

Reviewer 2: At first I did include block libraries in this section, but since it's not actually "UI" stuff, I've decided to leave it for your next section on rules. Block libraries are basically similar to Rule templates more so than UI pages/widgets.

Reviewer: I guess it depends on how you view it, but technically, they are exactly like widgets and pages. The code is in fact virtually the same. Block libraries are pure UI components that are only understood/handled by MainUI.

If you're thinking conceptually, it gets a bit blurrier, but there's no place I would naturally mention them when talking about rules. I guess what they "really are", are a kind of "scripting language libraries" with additional pre-made functionality.

Rule templates are universal and apply across scripting language/rule provider/storage syntax, block libraries apply only to Blockly and only in MainUI.

Reviewer 2: If you don't think they don't belong in the rule templates section, then a separate section might be better. I really don't think block libraries are "UI stuff". Users don't necessarily know nor care that they are _implemented_ as "UI components". They are not for presentation. They are for coding.

Reviewer: > They are not for presentation. They are for coding.

*In the UI*, using a form of "coding" that is only supported by the UI. Core has no idea what Blockly is.

But, I'll leave it to others to decide, it would just be so easy to mention it here along with the other almost similar components.

Author: I tend to put them besides rules and rule templates and not pages and widgets - I know the technical details, but I guess the actual implementation details aren't relevant in the blog post.

Key details of this new feature include:

- **Marketplace Integration:**<br/>
  Custom widgets can be easily adapted from the openHAB Marketplace for your configuration files.

>   You simply need to use the UID as the map key and nest the marketplace configuration underneath it.

Reviewer: I'm not so sure about this section. I have added support for both the "old" and "new" YAML format on the marketplace, so this won't always be a case of adding the UID as a map key.

An easier way might simply be to install the widget or block library from the marketplace, open them up on the "UI editor" and then copy/paste the "code" into your YAML files.

Reviewer 2: > An easier way might simply be to install the widget or block library from the marketplace, open them up on the "UI editor" and then copy/paste the "code" into your YAML files.

That step of installing it first might not be what most file-based users would prefer to do.

> I'm not so sure about this section. I have added support for both the "old" and "new" YAML format on the marketplace, so this won't always be a case of adding the UID as a map key.

Can you point me to where the changes are and the two marketplace formats that are now supported?

Reviewer: > That step of installing it first might not be what most file-based users would prefer to do.

That might be, but the only alternative is to make two different explanations for the two types of marketplace add-ons you'll find.

> Can you point me to where the changes are and the two marketplace formats that are now supported?

https://github.com/openhab/openhab-core/pull/5590

Author: Resolved by https://github.com/openhab/openhab-website/pull/564/commits/559a24dc54a51ac582ab074b58e9d5f365d7cdd5
